### PR TITLE
Fix zero length shape after  WcsGeom.cutout()

### PR DESCRIPTION
**Pull request by QRemy**
This fix a bug where WcsGeom.cutout() creates a geom with zero length shape.
In the test case added the problem appears when the width passed to `astropy.nddata.Cutout2D` is in degrees but not if it is in pixels.

### CODE REVIEW

File: gammapy/maps/wcs/tests/test_geom.py [hunk 1]
```diff
@@ -620,3 +620,13 @@ def test_wcs_geom_to_even_npix():
...
     assert geom_even.data_shape == (4, 4)
+
+
+def test_wcs_geom_no_zero_shape_cut():
+    geom = WcsGeom.create(skydir=(0, 2.5), binsz=(360, 5), width=(360, 180))
+    skydir = SkyCoord(110.0, 75.0, unit="deg", frame="icrs")
+
+    geom_cut = geom.cutout(skydir, width=5)
+
+    assert geom_cut.data_shape != (1, 0)
```
Comment (registerrier) [inline gammapy/maps/wcs/tests/test_geom.py:631]: why the two tests?

**Review by registerrier — DISMISSED** [review]
Thanks @QRemy. Looks good to me. I have just left a small inline comment.

Please rebase to avoid the black CI fails. Thanks!

File: gammapy/maps/wcs/tests/test_geom.py (hunk 1 above)
Comment (QRemy) [reply gammapy/maps/wcs/tests/test_geom.py:631]: I added this one such as it is more explicit that what we want to avoid in this test is the zero shape.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy!

**Review by adonath — APPROVED** [review]
Thanks @QRemy, no comments from my side...
